### PR TITLE
fix symfony deprecations for event dispatching

### PR DESCRIPTION
**Pull request by dmaicher**
## Subject

This is a follow-up for https://github.com/sonata-project/SonataAdminBundle/pull/5842.

Now that we dropped support for Symfony < 4.3 it simplifies the patch :blush: 

I am targeting this branch, because its BC.

## Changelog

```markdown
### Fixed
- deprecations for event dispatching

```

### CODE REVIEW

**Comment by greg0ire** [comment]
I know the contracts package is installed transitively, but please add it to the `require` section anyway, we need to pin it to one major version, to avoid bugs when the next major version is released.

File: composer.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
         "symfony/dependency-injection": "^4.3",
         "symfony/doctrine-bridge": "^4.3",
         "symfony/event-dispatcher": "^4.3",
+        "symfony/event-dispatcher-contracts": "^1.1",
```
Comment (greg0ire) [inline composer.json]: Sorry, I said one major version, but it is compatible with version 2, we should allow it. :sweat_smile:  So… is it?

Comment (dmaicher) [reply composer.json]: See https://github.com/symfony/event-dispatcher/blob/4.3/composer.json#L20

Comment (dmaicher) [reply composer.json]: We need to adjust it for Symfony 5 compat then at some point:

https://github.com/symfony/event-dispatcher/blob/5.0/composer.json#L20

Comment (greg0ire) [reply composer.json]: Ok, it can be done when working on allowing sf 5.

**Comment by greg0ire** [comment]
Thanks @dmaicher !

**Pull request merged**